src/components/common/AnalyticsTable.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { AppAction } from '@graasp/sdk';
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TableSortLabel,
+  Typography,
+} from '@mui/material';
+import katex from 'katex';
+import { useTranslation } from 'react-i18next';
+import { ActionData, AnalyticColumn, Order } from '@/types/table';
+import {
+  ANALYTIC_ROW_EQUATION_ID,
+  ANALYTIC_ROW_RESULT_ID,
+  ANALYTIC_ROW_CREATED_AT_ID,
+  ANALYTIC_ROW_MEMBER_ID,
+  buildAnalyticRowId,
+} from '@/config/selectors';
+import { PI_SYMBOL } from '@/config/constants';
+import { dateColumnFormatter } from '@/utils/action';
+
+interface Props {
+  columns: AnalyticColumn[];
+  rows: AppAction[];
```

Comment:
Using the generic type here: 
```suggestion
rows: AppAction<ActionData>[];
```
Should remove the casts later.